reference/test/matrix/dense_kernels.cpp

```diff
@@ -2154,13 +2154,95 @@ TYPED_TEST(Dense, SquareSubmatrixCanGatherRowsIntoDense)
     using T = typename TestFixture::value_type;
     auto exec = this->mtx5->get_executor();
     gko::Array<gko::int32> permute_idxs{exec, {1, 0}};
-    auto row_gathered = Mtx::create(exec, gko::dim<2>{2, 2}, 4);
+    auto row_collection = Mtx::create(exec, gko::dim<2>{2, 2}, 4);
 
     this->mtx5->create_submatrix({0, 2}, {1, 3})
-        ->row_gather(&permute_idxs, row_gathered.get());
+        ->row_gather(&permute_idxs, row_collection.get());
 
-    GKO_ASSERT_MTX_NEAR(row_gathered, l<T>({{2.0, 4.5}, {-1.0, -0.5}}), 0.0);
-    ASSERT_EQ(row_gathered->get_stride(), 4);
+    GKO_ASSERT_MTX_NEAR(row_collection, l<T>({{2.0, 4.5}, {-1.0, -0.5}}), 0.0);
+    ASSERT_EQ(row_collection->get_stride(), 4);
+}
+
+
+TYPED_TEST(Dense, NonSquareSubmatrixCanGatherRowsWithRowGatherable)
+{
+    // clang-format off
+    // {1.0, 3.0, 2.0},
+    // {0.0, 5.0, 0.0}
+    // clang-format on
+
```

Comment:
Do you need the empty line?
Also, please add that this is `mtx4` in the comments. It took me some time to realize that.
I also don't think you need to turn off clang-format for this:
```suggestion
    // mtx4: {{1.0, 3.0, 2.0}, {0.0, 5.0, 0.0}}
```

The same comment applies for the tests below with the same matrix as a comment.

Reply:
I delete them now.
I think they are shown to help checking test result, but it requires update when we change the initialization of mtx*